### ISSUE TITLE
Generated java files should be added to published source artifact

### ISSUE THREAD

**Issue by dpolivaev** [tbroyer/gradle-apt-plugin#78]
Currently if I generate artifact using 
~~~~
task sourceJar(type: Jar) {
	from sourceSets.main.allJava
	classifier "sources"
  }
  
publishing {
	publications {
		freeplaneEvents(MavenPublication) {
			from components.java
			artifactId 'freeplane-events'
			artifacts = [jar]
			artifact sourceJar {
				classifier "sources"
			}
		}
	}
}
~~~~
generated files are not added to freeplane-events-0.0.1-SNAPSHOT-sources.jar .
Therefore they can not be found when the jar is referenced from other projects.

How can I change the configuration?
Maybe the plugin should do it automatically?

Regards,
Dimitry

**Comment by dpolivaev** [comment]
As a work around I can use 
~~~~
task sourceJar(type: Jar) {
	from sourceSets.main.allJava, file('apt_generated')
	classifier "sources"
  }
~~~~

**Comment by tbroyer** [comment]
You should use `compileJava.options.annotationProcessorGeneratedSourcesDirectory` (or `compileJava.apt.generatedSourcesDir` if you use an older Gradle version; or possibly `sourceSets.main.output.generatedSourcesDir` added by this plugin) instead of the `file(…)`, but otherwise that's the way to go. You may want to use `allSource` rather than `allJava`. And you probably should add a task dependency to the `compileJava` task to make sure the code is compiled and sources are generated _before_ you actually package the source JAR.

I added a comment on https://github.com/gradle/gradle/issues/2300#issuecomment-371473196 about whether the generated sources directory should automatically be added to the `sourceSets.main.allJava`; but this has many side effects.

**Comment by tbroyer** [comment]
Fwiw, I just released 0.16 with this change.

**Comment by anuraaga** [comment]
Hi @tbroyer - what do you think about adding a setting to flip this on or off (default on is fine)? I've been wrangling with this change for a couple days and it's pretty tiring tracking down all the plugins that get affected by having generated code in `allJava` (broken javadoc, spotless, license format plugin, personally would much rather add the generated code in my `sourceJar` task manually as it's much simpler.

**Comment by tbroyer** [comment]
@anuraaga I'd suggest keeping 0.15 then. Given your feedback, maybe I'll just revert the change (it also "broke" IDE integrations, see #79)

**Comment by tbroyer** [comment]
Fwiw, I reverted this change and released v0.17 (which is, functionally, the same as v0.15)
